.github/workflows/build-and-test.yml

```diff
@@ -11,7 +11,7 @@ jobs:
 
       - uses: actions/setup-go@v2
         with:
-          go-version: '^1.16.6'
+          go-version: '~1.16'
```

Comment:
not sure if we want the Go version to be a moving target. Is this necessary for any of the other changes in this PR to work?

/cc @hectorj2f

Reply:
`^1.16.6` means `< 2.0`, `~1.16` means `< 1.17`, so actually I'm limiting the range rather then expand it. I made this change because the current latest Go 1.x release (1.18) is breaking the golangci-lint in test.